samples/aspnetcore/mvc/produces/src/ProducesMatcherPolicy/ProducesMatcherPolicy.cs

```diff
@@ -0,0 +1,256 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Formatters;
+using Microsoft.AspNetCore.Mvc.Formatters.Internal;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.AspNetCore.Routing.Matching;
+using Microsoft.Net.Http.Headers;
+
+namespace ProducesMatcherPolicy
+{
+    public class ProducesMatcherPolicy : MatcherPolicy, IEndpointComparerPolicy, INodeBuilderPolicy
+    {
+        internal const string Http404EndpointDisplayName = "404 HTTP Unsupported Accept Media Type";
```

Comment:
This should be 415 rather than 404

Reply:
I changed it to 404 after I looked up [415](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/415). The status is specific to the content type of the payload.

Neither 404 or 415 fit that well. Do you still want 415?

Reply:
If no [Produces] Content-Type matches the "Accept"-Header the Server SHOULD send a 406. As far as I mean the server also could send a different Content-Type Content, with the correct Header.


https://tools.ietf.org/html/rfc2616#page-100

Reply:
Sorry, sent an outdated RFC: 

>    A request without any Accept header field implies that the user agent>
   will accept any media type in response.  If the header field is
   present in a request and none of the available representations for
   the response have a media type that is listed as acceptable, the
   origin server can either honor the header field by sending a 406 (Not
   Acceptable) response or disregard the header field by treating the
   response as if it is not subject to content negotiation.


https://tools.ietf.org/html/rfc7231#page-38

Reply:
Looked up status codes and 406 is what we want https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/406